### PR TITLE
[#160196894]: Update user feedback feature 

### PR DESCRIPTION
**Pull request by nelmandela**
### What does this PR do?

  •Updates user feedback feature

### Description of the task to be completed:

  •Add `resolved` field
  •Add `feature_request` option

### How can this be manually tested?

  •Test `user feedback` endpoint

### Relevant PT stories:

[#160196894](https://www.pivotaltracker.com/story/show/160196894)

### Screenshots

screen shot 2018-09-05 at 11 13 54

### CODE REVIEW

File: core/models/user.py [hunk 1]
```diff
@@ -106,6 +108,7 @@ class UserFeedback(models.Model):
                                    choices=option,
```
Comment (collinmutembei) [inline core/models/user.py]: With the `max_length` argument on the `report_type` field set at 10 characters, a user **cannot** submit feedback of type `feature_request` as it is 15 characters in length. Update that value a number greater than that.

File: core/migrations/0028_auto_20180905_0927.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            model_name='userfeedback',
+            name='resolved',
+            field=models.BooleanField(default=False),
+        ),
+        migrations.AlterField(
+            model_name='userfeedback',
+            name='report_type',
+            field=models.CharField(choices=[('feedback', 'feedback'), ('bug', 'bug'), ('feature_request', 'feature_request')], max_length=10),
+        ),
+    ]
```
Comment (collinmutembei) [inline core/migrations/0028_auto_20180905_0927.py]: Regenerate the migration afresh after making the changes to the model, to have only one file.

File: core/models/user.py [hunk 3]
```diff
@@ -95,9 +95,11 @@ class UserFeedback(models.Model):
...
     FEEDBACK = "feedback"
     BUG = "bug"
+    FEATURE_REQUEST = "feature_request"
     option = (
         (FEEDBACK, "feedback"),
         (BUG, "bug"),
+        (FEATURE_REQUEST, "feature_request"),
     )
     reported_by = models.ForeignKey(User, on_delete=models.PROTECT)
     message = models.CharField(max_length=200, null=False)
```
Comment (collinmutembei) [inline core/models/user.py]: Also, update the `message` field to a [TextField](https://docs.djangoproject.com/en/2.1/ref/models/fields/#django.db.models.TextField) to accommodate reports that may exceed the 200 character limit currently set on the field.

File: core/models/user.py [hunk 4]
```diff
@@ -106,6 +108,7 @@ class UserFeedback(models.Model):
                                    choices=option,
                                    null=False)
     created_at = models.DateTimeField(auto_now_add=True, editable=False)
+    resolved = models.BooleanField(default=False)
 
     class Meta:
         verbose_name_plural = "User Feedback"
```
Comment (collinmutembei) [inline core/models/user.py]: Add tests for the `feature request` report type.

File: core/models/user.py [hunk 5]
```diff
@@ -95,9 +95,11 @@ class UserFeedback(models.Model):
     """ Stores user feedback data """
     FEEDBACK = "feedback"
     BUG = "bug"
+    FEATURE_REQUEST = "feature_request"
     option = (
         (FEEDBACK, "feedback"),
         (BUG, "bug"),
+        (FEATURE_REQUEST, "feature_request"),
```
Comment (collinmutembei) [inline core/models/user.py]: Rename value to `feature request`; drop the underscore.

File: core/models/user.py (hunk 4 above)
Comment (nelmandela) [reply core/models/user.py]: @collinmutembei i don't understand what we are supposed to test for here

Comment (nelmandela) [reply core/models/user.py]: @collinmutembei done, but coverage has decreased & i'm unable to locate the file causing the reduction

Comment (nelmandela) [reply core/models/user.py]: @collinmutembei done

File: api/tests/test_user_feedback_api.py [hunk 6]
```diff
@@ -89,3 +89,19 @@ def test_can_get_all_feedback(self, mock_verify_token):
...
+        data = {
+            "message": "Buy durable dongles",
+            "report_type": "feature request"
+        }
+        response = client.post(
+            self.feedback_url,
+            data=data,
+            HTTP_AUTHORIZATION="Token {}".format(self.token_user))
+
+        self.assertNotEqual(response.data, "resolved")
```
Comment (collinmutembei) [inline api/tests/test_user_feedback_api.py]: 🤔Of course `response.data` is not equal to "resolved". It is:
```python
{'reported_by': 'test4@site.com', 'message': 'Buy durable dongles', 'report_type': 'feature request', 'created_at': '2018-09-05T15:00:20.631259Z'}
```
You need to include the `resolved` field on serializer fields to check that its value is `false`.

Comment (nelmandela) [reply api/tests/test_user_feedback_api.py]: @collinmutembei done

File: api/serializers.py [hunk 7]
```diff
@@ -200,7 +200,7 @@ def to_representation(self, instance):
 class UserFeedbackSerializer(serializers.ModelSerializer):
     class Meta:
         model = UserFeedback
-        fields = ("reported_by", "message", "report_type", "created_at")
+        fields = ("reported_by", "message", "report_type", "created_at", "resolved")
         read_only_fields = ("reported_by",)
```
Comment (collinmutembei) [inline api/serializers.py]: Include the `resolved` field in the **read_only_fields** tuple as well and we're good to merge.

**Comment by collinmutembei** [comment]
@nelmandela Good to merge.

**Pull request merged**